internal/modules/overview/terminalviewer/terminalviewer.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (c) 2019 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package terminalviewer
+
+import (
+	"context"
+	"github.com/vmware-tanzu/octant/internal/terminal"
+	"github.com/vmware-tanzu/octant/pkg/log"
+	"github.com/vmware-tanzu/octant/pkg/store"
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ToComponent converts an object into a terminal component.
+func ToComponent(ctx context.Context, object runtime.Object, terminalManager terminal.Manager, logger log.Logger) (*component.Terminal, error) {
+	yv, err := new(ctx, object, terminalManager, logger )
+	if err != nil {
+		return nil, errors.Wrap(err, "create Terminal viewer")
+	}
+
+	return yv.ToComponent()
+}
+
+// Terminal Viewer is a terminal viewer for objects.
+type terminalViewer struct {
+	object runtime.Object
+	terminalManager terminal.Manager
+	context context.Context
+	logger log.Logger
+}
+
+// New creates an instance of TerminalViewer.
+func new(context context.Context, object runtime.Object, terminalManager terminal.Manager, logger log.Logger) (*terminalViewer, error) {
+	if object == nil {
+		return nil, errors.New("can't create Terminal view for nil object")
+	}
+
+	return &terminalViewer{
+		context: context,
+		object: object,
+		terminalManager: terminalManager,
+		logger: logger.With("Terminal Viewer", context),
+	}, nil
+}
+
+// ToComponent converts the Terminal Viewer to a component.
+func (tv *terminalViewer) ToComponent() (*component.Terminal, error) {
+	pod, err := getPod(tv)
+	if err != nil {
+		return nil, err
+	}
+
+	key, err:=  store.KeyFromObject(tv.object)
+	if err != nil {
+		return nil, err
+	}
+
+	container:= ""
+	if len(pod.Spec.Containers) > 0 {
+		container= pod.Spec.Containers[0].Name
```

Comment:
Are we always defaulting to the first container in the pod? What if that container was an init container?

Reply:
I think we can check if the container is an init container and skip it?

Reply:
I implemented the recommendation from Wayne but still seeing the above error sometimes.  Not sure exactly why, so we may need to address this again in future